src/integration-test/java/io/aiven/elasticsearch/gcs/GcsStoragePluginIT.java

```diff
@@ -0,0 +1,229 @@
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString())
+            .body(enableRepositoryRequest)
+            .execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+        }
+
+        try (final var response = EcrCurl.get(node, "_snapshot/backup").execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+            final Map<String, Object> content = response.getContent(EcrCurl.jsonParser());
+            assertEquals(
+                "{backup={settings={bucket_name=ples-test, base_path=" + BASE_PATH + "}, type=gcs-encrypted}}",
+                content.toString()
+            );
+        }
+
+    }
+
+    @Test
+    @Order(2)
+    void createSnapshot() throws Exception {
+        final var node = clusterRunner.getNode(0);
+
+        for (int i = 0; i < 10; i++) {
+            try (final var curlResponse = EcrCurl.post(node, "/" + INDEX + "/_doc/")
+                .header("Content-Type", "application/json")
+                .body("{\"id\":\"200" + i + "\",\"msg\":\"test 200" + i + "\"}")
+                .execute()) {
+                final Map<String, Object> content = curlResponse.getContent(EcrCurl.jsonParser());
+                assertNotNull(content);
+                assertEquals("created", content.get("result"));
+            }
+        }
+
+        try (final var response =
+                 EcrCurl.put(node, "/_snapshot/backup/snapshot_1?wait_for_completion=true")
+                     .execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+        }
+
+        final var metadataBlob = storage.get(BlobId.of(bucketName, BASE_PATH + "/" + GcsBlobStoreRepository.ENCRYPTION_METADATA));
+        assertNotNull(metadataBlob);
+        assertTrue(metadataBlob.exists());
+
+        final var listOfFilesIterator = listOfFileForBackupFolder();
+        assertNotNull(listOfFilesIterator);
+        assertTrue(countElementsInBackup(listOfFilesIterator) > 0);
+    }
+
+    @Test
+    @Order(3)
+    void restoreSnapshot() throws Exception {
+        clusterRunner.deleteIndex(INDEX);
+
+        assertFalse(clusterRunner.indexExists(INDEX));
+
+        final var node = clusterRunner.getNode(0);
+        try (final var response =
+                 EcrCurl.post(
+                     node,
+                     "/_snapshot/backup/snapshot_1/_restore?wait_for_completion=true"
+                 ).execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+        }
+
+        assertTrue(clusterRunner.indexExists(INDEX));
+    }
+
+    @Test
+    @Order(4)
+    void deleteRepository() throws Exception {
+        final var node = clusterRunner.getNode(0);
+
+        try (final var response = EcrCurl.delete(node, "_snapshot/backup").execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+        }
+
+        //The snapshots themselves are left untouched and in place
```

Comment:
Could you also test snapshot deletion? It should actually delete the snapshot files.

Reply:
my fault ... `DELETE _snapshot/backup` just remove repository, but keep content ... will use `POST /_snapshot/my_repository/_cleanup` instead